apis/python/setup.py

```diff
@@ -209,6 +209,9 @@ def run(self):
 # be at top level, outside any if-block
 # https://github.com/pypa/cibuildwheel/blob/7c4bbf8cb31d856a0fe547faf8edf165cd48ce74/cibuildwheel/projectfiles.py#L41-L46
 
+# Needed for Python 3.7 which anndata 0.9 doesn't support but we do
+anndata_version = "anndata < 0.9" if sys.version_info < (3, 8, 0) else "anndata"
```

Comment:
You can use platform-specific dependency specification and avoid this manual version detection:

https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#platform-specific-dependencies